src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -386,11 +386,16 @@ private async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Block
                 // Resync in case can't connect.
                 await this.ResyncAsync().ConfigureAwait(false);
             }
-            catch (ConsensusErrorException exception)
+            catch (ConsensusRuleException exception)
             {
                 this.logger.LogDebug("Peer's header is invalid. Peer will be banned and disconnected. Error: {0}.", exception.ConsensusError);
                 this.peerBanning.BanAndDisconnectPeer(peer.PeerEndPoint, this.connectionManager.ConnectionSettings.BanTimeSeconds, $"Peer presented invalid header, error: {exception.ConsensusError}.");
             }
+            catch (HeaderInvalidException)
```

Comment:
If we are going to pass the specific message to the exception we might want to write this to the log as well?